components/library/objfw/developer.p5m

```diff
@@ -0,0 +1,263 @@
+#
+# This file and its contents are supplied under the terms of the
+# Common Development and Distribution License ("CDDL"), version 1.0.
+# You may only use this file in accordance with the terms of version
+# 1.0 of the CDDL.
+#
+# A full copy of the text of the CDDL should have accompanied this
+# source.  A copy of the CDDL is also available via the Internet at
+# http://www.illumos.org/license/CDDL.
+#
+
+#
+# Copyright 2023 Jonathan Schleifer
+#
+
+set name=pkg.fmri value=pkg:/developer/objfw@$(IPS_COMPONENT_VERSION),$(BUILD_VERSION)
+set name=pkg.human-version value=$(HUMAN_VERSION)
+set name=pkg.summary value="$(COMPONENT_SUMMARY)"
+set name=info.classification value="$(COMPONENT_CLASSIFICATION)"
+set name=info.upstream-url value=$(COMPONENT_PROJECT_URL)
+set name=info.source-url value=$(COMPONENT_ARCHIVE_URL)
+set name=org.opensolaris.consolidation value=$(CONSOLIDATION)
+
+depend fmri=archiver/ofarc type=require
+depend fmri=network/ofdns type=require
+depend fmri=network/ofhttp type=require
+depend fmri=sysutils/ofhash type=require
+
+license LICENSE.QPL license=QPL
+license LICENSE.GPLv3 license=GPLv3
+license LICENSE.GPLv2 license=GPLv2
+
+<transform file -> add pkg.depend.bypass-generate libgcc_s\.so\.1>
```

Comment:
This is very very suspicious!

Reply:
Without this, it complains about not being able to resolve the dependency. Even adding gcc-10-compat-links as a dependency does not help. I was unable to find another solutions to this, as were the people in IRC.

Reply:
What exactly is the error message?

Reply:
```
/export/home/js/oi-userland/components/library/objfw/build/manifest-i386-developer.depend has unresolved dependency '
    depend type=require fmri=__TBD pkg.debug.depend.fullpath=lib/libgcc_s.so.1 \
        pkg.debug.depend.reason=usr/bin/objfw-new pkg.debug.depend.type=elf'.
/export/home/js/oi-userland/components/library/objfw/build/manifest-i386-library.depend has unresolved dependency '
    depend type=require fmri=__TBD pkg.debug.depend.fullpath=lib/libgcc_s.so.1 \
        pkg.debug.depend.reason=usr/lib/libobjfw.so.1.0 \
        pkg.debug.depend.type=elf'.
/export/home/js/oi-userland/components/library/objfw/build/manifest-i386-library.depend has unresolved dependency '
    depend type=require fmri=__TBD pkg.debug.depend.fullpath=lib/libgcc_s.so.1 \
        pkg.debug.depend.reason=usr/lib/libobjfwrt.so.1.0 \
        pkg.debug.depend.type=elf'.
/export/home/js/oi-userland/components/library/objfw/build/manifest-i386-library.depend has unresolved dependency '
    depend type=require fmri=__TBD pkg.debug.depend.fullpath=lib/libgcc_s.so.1 \
        pkg.debug.depend.reason=usr/lib/libobjfwtls.so.1.0 \
        pkg.debug.depend.type=elf'.
/export/home/js/oi-userland/components/library/objfw/build/manifest-i386-ofarc.depend has unresolved dependency '
    depend type=require fmri=__TBD pkg.debug.depend.fullpath=lib/libgcc_s.so.1 \
        pkg.debug.depend.reason=usr/bin/ofarc pkg.debug.depend.type=elf'.
/export/home/js/oi-userland/components/library/objfw/build/manifest-i386-ofhash.depend has unresolved dependency '
    depend type=require fmri=__TBD pkg.debug.depend.fullpath=lib/libgcc_s.so.1 \
        pkg.debug.depend.reason=usr/bin/ofhash pkg.debug.depend.type=elf'.
/export/home/js/oi-userland/components/library/objfw/build/manifest-i386-ofhttp.depend has unresolved dependency '
    depend type=require fmri=__TBD pkg.debug.depend.fullpath=lib/libgcc_s.so.1 \
        pkg.debug.depend.reason=usr/bin/ofhttp pkg.debug.depend.type=elf'.
```

Reply:
The problem is that your binaries are missing `RPATH`/`RUNPATH`.  You need to solve that otherwise users of this component could experience various strange issues in future (in addition to this one you are experiencing now).

Reply:
That sounds like a GCC / OpenIndiana bug then? If GCC links against `libgcc_s.so.1` but it requires a special rpath, then GCC should add that rpath via the default linker flags. `/usr/lib` cannot be in the rpath, that causes an error (this is what happens without passing `--disable-rpath` to configure), but that's where the symlink lives. I also don't see other packages adding an rpath for gcc? Hardcoding `/usr/gcc/10/lib/amd64` also sounds like a bad idea that would entirely defeat the purpose of depending on `$(GCC_RUNTIME_PKG)`.

So I think the real solution here would be that the packaging system should provide the proper `LDFLAGS` to be picked up that contain such an rpath?

Reply:
I manually added an rpath for GCC now. This feels dirty and wrong, but the allowed me to remove the transform.

Reply:
How other components do this?   Isn't a problem (if there is really a problem) in `objfw`?  Maybe there is a bug in GCC or in the OpenIndiana build framework.  PRs are always welcome to address such issues.

Reply:
I couldn't find anything else that pulled in just libgcc_s.so.1 and nothing else - only together with C++ stuff. Maybe some of that C++ stuff causes the rpath to be added correctly?

Reply:
In any case: The pkg5 file references gcc 10 already anyway. So if this gets updated, things will need to be manually changed anyway, so probably this hack isn't too bad.

Reply:
> I couldn't find anything else that pulled in just libgcc_s.so.1 and nothing else - only together with C++ stuff. Maybe some of that C++ stuff causes the rpath to be added correctly?

What about `/usr/bin/augmatch` or `/usr/bin/cargo-fmt`, for example?

Reply:
> In any case: The pkg5 file references gcc 10 already anyway. So if this gets updated, things will need to be manually changed anyway, so probably this hack isn't too bad.

It would be better if there is no manual change needed in future.

Reply:
I figured out the difference: `libgcc_s.so.1` is not being added here because GCC pulled it in itself. It's being added because configure checks which libraries are required for unwinding and finds `libgcc_s`, adding an explicit `-lgcc_s`. For e.g. C++, GCC knows that `libgcc_s` is required and adds it itself. But for ObjC, it doesn't know that it's linking ObjC code in the end.

Reply:
Also found a way to get the right directory without hardcoding versions.